### PR TITLE
Sequential task fixes

### PR DESCRIPTION
**Pull request by phgross**
This PR provides some fixes for the sequential task process feature:
- Fixed automatic start of a next task inside a sequential task process.
- Only show "add task to process" link, if next task is not yet started.
- Fix adding sequential task process on first position.

This ensures that only one task is open in a process.

For https://4teamwork.atlassian.net/browse/CA-1429

## Checklist (Must have)
- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)
- [x] Works also for the new UI

### CODE REVIEW

**Review by njohner — COMMENTED** [review]
Fixes look good, but if possible using the same code for the new and classic UI would be nice.
Also I tested the branch locally and two of the problems I'd found seem to be still present:
- skip a task and then reopen it -> on the main task in the sequential task representation, it is still represented as skipped
- Skip a task, resolve the previous task in the sequence. It opens the next one. Then reopen the skipped task. Two open tasks. But this one is probably unimportant.

**Comment by phgross** [comment]
Hi @niklaus I've fixed the issue with `skipped-open` transition. There was no TransitionExtender registered for this transition. This leads to two "open" tasks inside a sequential process, but to no other problems or issues.

I also tried to use the same code for the api (new UI) and the classic UI, but currently the complete task overview is based on the SQL Task and not the real object. So I don't think its worth the risk to refactor the TaskOverview or do unneccessary SQL queries in the API Endpoint.

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -13,6 +13,7 @@ Changelog
 
 - Fix support in TransitionExtender for RelationChoice fields. [phgross]
 - Allow any authenticated users to use the REST API. [phgross]
+- Fix adding sequential task process on first position. [phgross]
```
Comment (njohner) [inline docs/HISTORY.txt]: Move it to 2021.6.0

File: docs/HISTORY.txt [hunk 2]
```diff
@@ -13,6 +13,7 @@ Changelog
 
 - Fix support in TransitionExtender for RelationChoice fields. [phgross]
 - Allow any authenticated users to use the REST API. [phgross]
+- Only show "add task to process" link, if next task is not yet started. [phgross]
```
Comment (njohner) [inline docs/HISTORY.txt]: This one too

File: docs/HISTORY.txt [hunk 3]
```diff
@@ -13,6 +13,7 @@ Changelog
 
 - Fix support in TransitionExtender for RelationChoice fields. [phgross]
 - Allow any authenticated users to use the REST API. [phgross]
+- Fixed automatic start of a next task inside a sequential task process. [phgross]
```
Comment (njohner) [inline docs/HISTORY.txt]: and this one

File: docs/HISTORY.txt [hunk 4]
```diff
@@ -4,6 +4,9 @@ Changelog
 2021.6.0 (unreleased)
 ---------------------
 
+- Fixed automatic start of a next task inside a sequential task process. [phgross]
+- Only show "add task to process" link, if next task is not yet started. [phgross]
+- Fix adding sequential task process on first position. [phgross]
```
Comment (njohner) [inline docs/HISTORY.txt:9]: ah ok 👍

**Review by njohner — APPROVED** [review]
LGTM!

**Pull request merged**